modules/proxy-server/proxy-user-data.sh

```diff
@@ -9,7 +9,12 @@ sed -i 's|SERVER_NAME|${server_name}|g' /tmp/templates/ca.crt.tpl
 sed -i 's|SERVER_NAME|${server_name}|g' /tmp/templates/server.crt.tpl
 sed -i 's|SERVER_NAME|${server_name}|g' /tmp/templates/server.key.tpl
 
-consul-template -once -config="/tmp/templates/consul-template-config.hcl" -vault-addr=${vault_address} -vault-token=${vault_token}
+VAULT_TOKEN=$(curl --insecure -X POST "${vault_address}/v1/auth/aws/login" -d '{"role":"vetservices","pkcs7":"'$(curl -s http://169.254.169.254/latest/dynamic/instance-identity/pkcs7 | tr -d '\n')'","nonce":"5defbf9e-a8f9-3063-bdfc-54b7a42a1f95"}' | jq '.auth.client_token' | tr -d '"')
+
+echo "<<DEBUG>>"
+echo "[consul-template -once -config="/tmp/templates/consul-template-config.hcl" -vault-addr=${vault_address} -vault-token=$${VAULT_TOKEN}]"
+
+consul-template -once -config="/tmp/templates/consul-template-config.hcl" -vault-addr=${vault_address} -vault-token=$${VAULT_TOKEN}
```

Comment:
Shouldn't this be `${VAULT_TOKEN}` instead of `$${VAULT_TOKEN}`? The debug statement uses `$$` because it doesn't want to output the actual value right?

Reply:
The curl statement has to fetch the vault token and set the shell variable for the script to use it in the consul-template statement.

The double $ sign keeps it as a shell variable, so that terraform won't try to interpolate it